webrtc/README.md

```diff
@@ -0,0 +1,271 @@
+# WebRTC
+
+| Lifecycle Stage | Maturity      | Status | Latest Revision |
+|-----------------|---------------|--------|-----------------|
+| 1A              | Working Draft | Active |                 |
+
+Authors: [@mxinden]
+
+Interest Group: [@marten-seemann]
+
+<!-- markdown-toc start - Don't edit this section. Run M-x markdown-toc-refresh-toc -->
+**Table of Contents**
+
+- [WebRTC](#webrtc)
+    - [Motivation](#motivation)
+    - [Requirements](#requirements)
+    - [Addressing](#addressing)
+    - [Connection Establishment](#connection-establishment)
+        - [Browser to public Server](#browser-to-public-server)
+            - [Open Questions](#open-questions)
+        - [Browser to Browser](#browser-to-browser)
+            - [Open Questions](#open-questions-1)
+    - [Connection Security](#connection-security)
+        - [Open Questions](#open-questions-2)
+    - [Multiplexing](#multiplexing)
+    - [General Open Questions](#general-open-questions)
+    - [Previous, ongoing and related work](#previous-ongoing-and-related-work)
+- [FAQ](#faq)
+
+<!-- markdown-toc end -->
+
+
+## Motivation
+
+1. **No need for valid TLS certificates.** Enable browsers to connect to public
+   server nodes without those server nodes providing a TLS certificate within
+   the browsers trustchain. Note that we can not do this today with our
+   Websocket transport.
+
+2. **Hole punching in the browser**: Enable two browsers or a browser and a
+   non-public server node to connect.
+
+## Requirements
+
+- Loading a remote nodes certificate into ones browser trust-store is not an
+  option, i.e. doesn't scale.
+
+- No dependency on central STUN and/or TURN servers.
+
+## Addressing
+
+WebRTC multiaddresses are composed of an IP and UDP address component, followed
+by `/webrtc` and a multihash of the certificate that the node uses.
+
+Examples:
+- `/ip4/1.2.3.4/udp/1234/webrtc/certhash/<hash>/p2p/<peer-id>`
```

Comment:
Following up on https://github.com/multiformats/go-multiaddr/pull/179#issuecomment-1172578231 (//CC @marten-seemann):

@melekes @ckousik do you agree with the above, namely to split the WebRTC addressing scheme into a `/webrtc` and a `/certhash` component?

Reply:
Will we be able to reuse `certhash` for webtransport?

Here are the supported hash functions: https://datatracker.ietf.org/doc/html/rfc8122#section-5

```
"sha-1" / "sha-224" / "sha-256" /
                             "sha-384" / "sha-512" /
                             "md5" / "md2" / token
```

Reply:
> Will we be able to reuse `certhash` for webtransport?

That would be the goal, yes.

> Here are the supported hash functions: https://datatracker.ietf.org/doc/html/rfc8122#section-5
> 
> ```
> "sha-1" / "sha-224" / "sha-256" /
>                              "sha-384" / "sha-512" /
>                              "md5" / "md2" / token
> ```

If we encode the fingerprint as a multihash, there being multiple supported hash functions should not be an issue.

Reply:
To be even more precise, it's a _multibase encoded_ multihash.

That allows implementations to choose the base they want to encode the multihash in, e.g. they could use (any variant of) base32, base58, base64, or even emoji-base.

Reply:
Discussion moved to https://github.com/multiformats/multiaddr/pull/130. Resolving here.